psi4/src/core.cc

```diff
@@ -70,6 +70,66 @@ using namespace psi;
 namespace py = pybind11;
 using namespace pybind11::literals;
 
+#ifdef USING_BrianQC
+#include <use_brian_wrapper.h>
+#include <brian_module.h>
+#include <brian_macros.h>
+
+bool brianEnableEnvFound = false;
+bool brianEnableEnvValue = false;
+
+BrianCookie brianCookie = 0;
+bool brianEnableDFT = true;
+brianInt brianRestrictionType = 0;
+bool brianCPHFFlag = false;
+bool brianCPHFLeftSideFlag = false;
+
+void checkBrian() {
+    brianBool err = brianAPIGetError(&brianCookie);
+    if (err) {
+        throw PSIEXCEPTION("BrianQC error detected");
+    }
+}
+
+void brianInit() {
+    if (brianCookie != 0) {
+        throw PSIEXCEPTION("Attempting to reinitialize BrianQC without releasing it first");
+    }
+    
+    brianInt brianAPIVersionOfHost = BRIAN_API_VERSION;
+    brianInt hostID = BRIAN_HOST_PSI4;
+    brianCookie = brianAPIInit(&brianAPIVersionOfHost, &hostID);
+    checkBrian();
+    outfile->Printf("BrianQC initialization successful\n");
+}
+
+void brianRelease()
+{
+    if (brianCookie == 0) {
+        throw PSIEXCEPTION("Attempting to release the BrianQC module when it hasn't been initialized\n");
+    }
+    
+    outfile->Printf("Releasing the BrianQC module\n");
+    brianAPIRelease(&brianCookie);
+    brianCookie = 0;
+}
+
+void handleBrianOption(bool value) {
+    if (brianEnableEnvFound) {
+        outfile->Printf("BRIANQC_ENABLE option found, but overridden by BRIANQC_ENABLE environment variable\n");
+    } else {
+        outfile->Printf("BRIANQC_ENABLE option found, checking value\n");
+        if (value && (brianCookie == 0)) {
+            outfile->Printf("BRIANQC_ENABLE option set to true, initializing BrianQC\n");
+            brianInit();
+        } else if (!value && (brianCookie != 0)) {
+            outfile->Printf("BRIANQC_ENABLE option set to false, releasing BrianQC\n");
+            brianRelease();
+        }
```

Comment:
I think toggling brianqc/pure-cpu control still needs working out (and if you want a conference call with me to discuss this easier, I'm glad to do so), but since there's no immediate users to disrupt, that can be in a follow-up PR.

Reply:
It would be good to discuss this issue over a video call, since I'm still not sure which potential solution is preferable. In the meantime, I will create a separate internal variable to enable and disable BrianQC, so that enabling is not tied to whether the cookie is initialized.

Reply:
sounds good. will switch to email, but I can do today or M–W of next week.

Reply:
From discussions today, we're thinking of two controls:
* an envvar to toggle whether BrianQC library (and GPUs) even gets initialized (takes about 10 s). 
* a psi4 option to toggle whether to use BrianQC for next jobs so that in an input one can run DFT w/ and w/o GPUs.
* BrianQC devs' choice about whether initialization is either once per `import psi4` at the same time as psi4 initialization or per-job where BrianQC is to be used.

They're going to talk this over and it can either go into this PR or a subsequent one, depending on review timings.